src/module_wrap.cc

```diff
@@ -321,6 +321,61 @@ static Local<Object> createImportAttributesContainer(
   return attributes;
 }
 
+void ModuleWrap::GetModuleRequestsSync(
```

Comment:
I think refactoring of the async path can be left as a TODO, no? (though I have already tried very hard not to leave too many TODO comments and just focus on getting require(esm) implemented in this PR..)

Reply:
Yeah, it can be done in a separate work. I found it surprising that `ModuleWrap::Link` unnecessarily makes the synchronous work harder and there is a `ModuleWrap::GetStaticDependencySpecifiers` to get module requests as well, just in specifiers only.

Reply:
Yup, I was surprised too, and tried hard not to refactor all the weird stuff that I found.... ;)